### ISSUE TITLE
R Websocket client does not connect from within R jupyter notebook

### ISSUE THREAD

**Issue by sheunaluko** [IRkernel/IRkernel#649]
Hi, thank you for your time.  

**Environment verification**
I installed R on OSX using the official instruction and verified the packages c('IRkernel', 'IRdisplay', 'repr') are the newest versions. 

**Context of Issue**
I am attempting to build an interactive graphing element within the R jupyter notebook environment. The graph is displayed inside an iframe in a cell, and the iframe javascript code connects to a websocket server on localhost. I am attempting to connect to the same websocket server using the websocket client library [https://rstudio.github.io/websocket/index.html](url), but the connection does not appear to be made (with no error message or debug info that I can see). 

**Steps to reproduce the problem:** 

1) Start a websocket server on localhost:10001  (I used nodejs script shown below) 

```
const WebSocket   = require('ws');
let port = 10001 ; 
const wss = new WebSocket.Server({ port: port });
wss.on('connection', function (ws) {
	console.log("Received ws connection.")	
    ws.on('message', console.log ) 
}) 

console.log("Waiting for connections on port: " + port) 

//To run this file save it to main.js then run "node main.js" from command line 
```

2) Install R websocket library -> `install.packages("websocket")`
3) Run "jupyter notebook" and select the R kernel 
4) Enter the following code in a jupyter cell and execute it

```
library("websocket")
ws <- WebSocket$new("ws://127.0.0.1:10001/") 
```

**Expected result:** 
The nodejs websocket server script should log "Received ws connection." but instead nothing happens. 

On the other hand, opening an R console from the command line and entering the exact same code from (4) above will result in the connection being made and the log being seen. 

**Is there a reason that the websocket client library just does not work inside of the R jupyter notebook?** I tried running jupyter in debug mode and could not gain any insights.  

Any help is much appreciated!

**Comment by flying-sheep** [comment]
True, also on Linux, but I don’t know why it’s happening. Even using `accessLogChannels='all'` and `ws$onError`, there’s no message. `ws$readyState()` just says “connecting”.

It *might* have to do with IRkernel using ZMQ to communicate with Jupyter, not sure. I’m not experienced in using Wireshark or so to debug packages, we should find someone to help.
